internal/cli/cmd/apply/apply.go

```diff
@@ -19,8 +19,54 @@ import (
 	kargosvcapi "github.com/akuity/kargo/pkg/api/service/v1alpha1"
 )
 
+type applyOptions struct {
+	*option.Option
+
+	Filenames []string
+}
+
+// newApplyOptions creates a new options struct.
+func newApplyOptions(opt *option.Option) *applyOptions {
+	return &applyOptions{Option: opt}
+}
+
+// AddFlags adds the flags for the apply options to the provided command.
+func (o *applyOptions) AddFlags(cmd *cobra.Command) {
+	o.PrintFlags.AddFlags(cmd)
+
+	// TODO: Factor out server flags to a higher level (root?) as they are
+	//   common to almost all commands.
+	option.InsecureTLS(cmd.PersistentFlags(), o.Option)
+	option.LocalServer(cmd.PersistentFlags(), o.Option)
+
+	option.Filenames(cmd.Flags(), &o.Filenames, "Filename or directory to use to apply the resource(s)")
+
+	if err := cmd.MarkFlagRequired(option.FilenameFlag); err != nil {
```

Comment:
These are new since the last time I went deep on Cobra. I am super excited about how much this cleans things up.